src/test/java/TextEditorTest.java

```diff
@@ -0,0 +1,22 @@
+import org.junit.jupiter.api.Test;
+
+class TextEditorTest {
+
+    @Test
+    void givenTextEditor_whenAddTextSaveAddMoreAndUndo_thenSavecStateRestored() {
```

Comment:
don't exactly love the name of this method but I do very much appreciate the content 😂